pilot/pkg/controller/workloadentry/workloadentry_controller.go

```diff
@@ -239,9 +272,69 @@ func shouldCleanupEntry(wle config.Config) bool {
 	return true
 }
 
-func setConnectMeta(c *config.Config, controller string, con *Connection) {
-	c.Annotations[WorkloadControllerAnnotation] = controller
-	c.Annotations[ConnectedAtAnnotation] = con.Connect.Format(timeFormat)
+func (c *Controller) cleanupEntry(wle config.Config) {
+	if err := c.cleanupLimit.Wait(context.TODO()); err != nil {
+		log.Errorf("error in WorkloadEntry cleanup rate limiter: %v", err)
```

Comment:
I have this in one of my PRs.. but we should have `returns` in the error cases here.. otherwise the successful message gets logged which can be confusing.

Reply:
I donot touch this code, it is same with before. I think no error should happen as the cleanupLimit is set right

Reply:
Fixed, this was a coding regulation issue